### DISCUSSION TITLE
Print e-mails to stdout

### DISCUSSION THREAD

**Discussion by dbrgn** [dani-garcia/vaultwarden#1458] [Ideas]
For a single-user setup, I'd prefer not to configure a SMTP server. However, certain things (like changing the e-mail address) require e-mail confirmation by default. Invitations generate e-mails by default.

Would a mode be possible, where e-mails are printed to stdout, instead of being sent via SMTP? Then I could get the e-mail body from my server logs.

**Comment by BlackDex** [comment]
I would probably use something like mailcatcher, mailhog or similar.
There are docker images available for it. Don't ask me which one to use, i have no idea.

**Reply by dbrgn** [reply]
Yeah, but that adds more complexity to the setup, not less 🙂

Would a pull request with such a feature be accepted? After looking at the sources, I think it should be doable with only a few lines of code.

**Reply by jjlin** [reply]
This seems like a super niche request. What's the big deal with setting up email? That's useful for other features too.

**Comment by BlackDex** [comment]
I'm not really fond of outputting that kind of info to stdout or stderr.
It would also add an extra flag to the already crowded config.

And, it's not something asked everyday.
Also, i rejected a PR for adding a bcc field which is also something not requested ever except once.

In my opinion, if you have bitwarden_rs running in docker, just run MailHog alongside it. It even gives you a nice web interface. And no hassel to set-it-up at all.

**Reply by dbrgn** [reply]
It would not require an option, it could simply be enabled if no SMTP host is configured.

In the end, its your choice as maintainers, so if you'd prefer not to have this feature, I'd accept that of course.

**Comment by BlackDex** [comment]
That would mean that everyone who do not have SMTP enabled would get this, which is not an option of course.

Besides that, it's just not something i want into the logs.
